metapool/plate.py

```diff
@@ -214,3 +217,198 @@ def _plate_position(well):
         return '2'
     elif row % 2 == 1 and col % 2 == 1:
         return '1'
+
+
+def dilute_gDNA(plate_df, threshold=15):
+    """
+        Reads a plate_df and returns plate_df w/1:10 diluted gDNA samples.
+        :param plate_df: processing-plate Pandas dataframe
+        :param threshold: Upper gDNA concentration threshold.
+        :return: Pandas DataFrame
+               If dilution was needed, gDNA concentrations will be adjusted.
+               'Diluted' column will be added and contents set to 'True'.
+    """
+    if 'Diluted' in plate_df.columns:
+        # function has already been applied to data
+        warnings.warn('Dilution operation was already performed')
+        return plate_df
+
+    # return a copy of the input data. Do not overwrite the input data by
+    # default.
+    df = plate_df.copy()
+    df['Diluted'] = True
+
+    df['Project Plate'] = df['Project Plate'].astype(str) + '_diluted'
+    df['Compressed Plate Name'] = df['Compressed Plate Name'] + '_dilution'
+    df['Sample DNA Concentration'] = df['Sample DNA Concentration'] / 10
+
+    # Picking diluted samples for normalization
+    diluteds = (df.loc[df['Sample DNA Concentration'] > (threshold / 10)])
+    undiluteds = plate_df.loc[~plate_df['Sample'].isin(diluteds['Sample'])]
+
+    return pd.concat([diluteds, undiluteds]).sort_index()
+
+
+def requires_dilution(plate_df, threshold=15, tolerance=0.05):
+    """
+        Reads a plate_df and determines whether or not it needs diluting.
+        :param plate_df: processing-plate Pandas dataframe
+        :param threshold: Upper gDNA concentration threshold.
+        :param tolerance: Percentage (0.0-1.0) allowed outside threshold.
+        :return: True if plate requires dilution. False otherwise.
+        """
+    # generate a Series of Booleans, based on whether each concentration falls
+    # within the threshold or not.
+    within_threshold = plate_df['Sample DNA Concentration'] > threshold
+    pct_of_total = plate_df.loc[within_threshold].shape[0] / plate_df.shape[0]
+
+    if pct_of_total > tolerance:
```

Comment:
should this be `return pct_of_total > tolerance`?

Reply:
works for me. :)